lobsterpy/featurize/core.py

```diff
@@ -0,0 +1,829 @@
+# Copyright (c) lobsterpy development team
+# Distributed under the terms of a BSD 3-Clause "New" or "Revised" License
+
+"""
+This module defines classes to featurize Lobster data ready to be used for ML studies
+"""
+
+from __future__ import annotations
+import gzip
+import json
+import os
+import warnings
+from pathlib import Path
+from typing import List, Tuple
+from collections import namedtuple
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+from mendeleev import element
+from pymatgen.core.structure import Structure
+from pymatgen.io.lobster import Charge, Icohplist, MadelungEnergies
+from pymatgen.electronic_structure.cohp import CompleteCohp
+from pymatgen.electronic_structure.core import Spin
+from scipy.integrate import trapezoid
+from lobsterpy.cohp.analyze import Analysis
+
+warnings.filterwarnings("ignore")
+
+
+class FeaturizeLobsterpy:
+    """
+    class to featurize lobsterpy data
+
+    Args:
+        path_to_json: path to lobster lightweight json
+        bonds: "all_bonds" or "cation_anion_bonds"
+    Attributes:
+        get_df: returns a pandas dataframe with relevant icohp statistical data as columns from
+        lobsterpy automatic bonding analysis
+
+    """
+
+    def __init__(
+        self,
+        path_to_lobster_calc: str | None = None,
+        path_to_json: str | None = None,
+        bonds: str = "all_bonds",
+    ):
+        self.path_to_json = path_to_json
+        self.path_to_lobster_calc = path_to_lobster_calc
+        self.bonds = bonds
+
+    def get_df(self, ids: str | None = None) -> pd.DataFrame:
+        """
+        This function featurizes LobsterPy condensed bonding analysis data from
+        lobster lightweight json.gz files
+
+        Returns:
+            Returns a pandas dataframe with lobsterpy icohp statistics
+
+        """
+        if self.path_to_json and not self.path_to_lobster_calc:
+            # read the lightweight lobster json files using read_lobster_lightweight_json method
+            data = self._read_lobster_lightweight_json()
+            if not ids:
+                ids = Path(self.path_to_json).name.split(".")[0]
+
+        elif self.path_to_lobster_calc and not self.path_to_json:
+            # get lobsterpy condensed bonding analysis data using get_lobsterpy_cba_dict method
+            data = self._get_lobsterpy_cba_dict()
+
+            if not ids:
+                ids = Path(self.path_to_lobster_calc).name
+
+        else:
+            raise ValueError(
+                "Please provide either path to lightweight lobster jsons or path to lobster calc"
+            )
+        # define a pandas dataframe
+        df = pd.DataFrame(index=[ids])
+
+        icohp_mean = []
+        icohp_sum = []
+        bond = []
+        antibond = []
+        # extract lobsterpy icohp related data for bond type specified
```

Comment:
Could you add a comment on what your statistical data means here? You might also want to mention that there will be differences between "all" and "cation-anion" mode, as in "all", the bonds will come up twice.